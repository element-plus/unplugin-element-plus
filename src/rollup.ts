/**
 * This entry file is for Rollup plugin.
 *
 * @module
 */

import unplugin from './index'

/**
 * Rollup plugin
 *
 * @example
 * ```ts
 * // rollup.config.js
 * import ElementPlus from 'unplugin-element-plus/rollup'
 *
 * export default {
 *   plugins: [ElementPlus()],
 * }
 * ```
 */
const rollup = unplugin.rollup as typeof unplugin.rollup
export default rollup
export { rollup as 'module.exports' }
