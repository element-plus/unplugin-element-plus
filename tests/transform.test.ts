/* eslint-disable unicorn/prefer-string-replace-all */
import { resolve } from 'node:path'
import glob from 'fast-glob'
import { describe, expect, it } from 'vitest'
import { rollup } from 'rollup'
import plugin from '../src/rollup'

async function getCode(file: string, plugin: any) {
  const bundle = await rollup({
    input: [file],
    plugins: [plugin],
    external: () => true,
  })
  const output = await bundle.generate({ format: 'esm' })
  return output.output
    .map((file) => {
      if (file.type === 'chunk') {
        return file.code
      } else {
        return file.fileName
      }
    })
    .join('\n')
}

describe('transform', () => {
  describe('fixtures', async () => {
    const root = resolve(__dirname, '..')
    const files = await glob('tests/fixtures/*.{vue,js,ts}', {
      cwd: root,
      onlyFiles: true,
    })

    for (const file of files) {
      describe(file.replace(/\\/g, '/'), () => {
        const filepath = resolve(root, file)

        for (const useSource of [true, false]) {
          it(`useSource = ${useSource}`, async () => {
            const code = await getCode(
              filepath,
              plugin({
                useSource,
                ignoreComponents: ['AutoResizer'],
              })
            )
            expect(code).toMatchSnapshot()
          })
        }
      })
    }
  })
})
