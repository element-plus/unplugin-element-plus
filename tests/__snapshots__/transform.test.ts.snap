// Vitest Snapshot v1

exports[`transform > fixtures > tests/fixtures/basic.ts > useSource = false 1`] = `
"import { ElButton, ElCol } from 'element-plus';
import 'element-plus/es/components/button/style/css';
import 'element-plus/es/components/col/style/css';

console.log(ElButton, ElCol);
"
`;

exports[`transform > fixtures > tests/fixtures/basic.ts > useSource = true 1`] = `
"import { ElButton, ElCol } from 'element-plus';
import 'element-plus/es/components/button/style/index';
import 'element-plus/es/components/col/style/index';

console.log(ElButton, ElCol);
"
`;

exports[`transform > fixtures > tests/fixtures/comments.ts > useSource = false 1`] = `
"import { ElButton, ElCol } from 'element-plus';
import 'element-plus/es/components/button/style/css';
import 'element-plus/es/components/col/style/css';

console.log(ElButton, ElCol);
"
`;

exports[`transform > fixtures > tests/fixtures/comments.ts > useSource = true 1`] = `
"import { ElButton, ElCol } from 'element-plus';
import 'element-plus/es/components/button/style/index';
import 'element-plus/es/components/col/style/index';

console.log(ElButton, ElCol);
"
`;

exports[`transform > fixtures > tests/fixtures/ignored.ts > useSource = false 1`] = `
"import { ElAutoResizer } from 'element-plus';
import 'element-plus/es/components/button/style/css';

console.log(ElAutoResizer);
"
`;

exports[`transform > fixtures > tests/fixtures/ignored.ts > useSource = true 1`] = `
"import { ElAutoResizer } from 'element-plus';
import 'element-plus/es/components/button/style/index';

console.log(ElAutoResizer);
"
`;

exports[`transform > fixtures > tests/fixtures/import-alias.ts > useSource = false 1`] = `
"import { ElButton, ElCard } from 'element-plus';
import 'element-plus/es/components/button/style/css';
import 'element-plus/es/components/card/style/css';

console.log(ElButton, ElCard);
"
`;

exports[`transform > fixtures > tests/fixtures/import-alias.ts > useSource = true 1`] = `
"import { ElButton, ElCard } from 'element-plus';
import 'element-plus/es/components/button/style/index';
import 'element-plus/es/components/card/style/index';

console.log(ElButton, ElCard);
"
`;
